cleangoldenami/main.go

```diff
@@ -58,23 +60,60 @@ func main() {
 				log.Fatalf("error fetching images for region %v: %v", regionName, err)
 			}
 
-			if imageCount := len(images); imageCount < quota {
+			if imageCount := len(images); (quota - imageCount) >= desiredImageCapacity {
 				if *verbose {
 					fmt.Printf("Region %v is under quota. (Images: %v, Quota: %v)\n", regionName, imageCount, quota)
 				}
 			} else {
-				imageToDelete, err := getOldestImage(images)
-				if err != nil {
-					log.Fatalf("error determining which image to delete in region %v: %v", regionName, err)
-				}
-				if *dryRun {
-					fmt.Printf("Region %v is at quota (%v) - would delete %v\n", regionName, quota, *imageToDelete.ImageId)
-				} else {
-					err := deregisterImage(ec2Client, imageToDelete)
+				numOfImagesToDelete := desiredImageCapacity - (quota - imageCount)
+
+				arm64Images := filterImages(images, "arm64", "rhel-arm64")
+				legacyx86Images := filterImages(images, "x86_64", "legacy-x86_64")
+				x86Images := filterImages(images, "x86_64", "rhel-x86_64")
+
+				var imagesToDelete []ec2Types.Image
+
+				for !imageDeletionCheck(imagesToDelete, numOfImagesToDelete, arm64Images, legacyx86Images, x86Images) {
```

Comment:
What are your thoughts on deleting the `imageDeletionCheck` function altogether and changing this as follows? Would that miss any cases that I'm not thinking of?
```suggestion
				for len(imagesToDelete) < numOfImagesToDelete {
```

Reply:
So, that approach won't handle cases where deletion is necessary but there is only one image for each type. I've updated the for loop to use shouldDeleteImages(imagesToDelete, numOfImagesToDelete, arm64Images, legacyx86Images, x86Images).

If you still think it would be better to use for len(imagesToDelete) < numOfImagesToDelete, I can simply add another conditional statement within the for loop to ensure that there are at least two images for the image type that needs to be deleted.

Reply:
Ah, that makes sense. Good call.

Summarizing our slack conversation here - let's have _some_ way to handle that as a special case being that it would require manual user action to resolve.